scripts/setup/nrfconnect/update_ncs.py

```diff
@@ -0,0 +1,105 @@
+parser.add_argument("-u", "--update", help="Update your nRF Connect SDK to currently recommended version.", action="store_true")
+args = parser.parse_args()
+
+chip_root = os.path.dirname(os.path.abspath(__file__)).replace("scripts/setup/nrfconnect", "")
+
+def get_repository_commit_sha(repository_location):
+    command = "git -C " + repository_location + " rev-parse HEAD"
+    process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    if process.returncode:
+        raise subprocess.CalledProcessError(process.returncode, command, output=stdout, stderr=stderr)
+    else:
+        return subprocess.CompletedProcess(process.args, process.returncode, stdout, stderr).stdout.decode('ascii').strip()
+
+def update_ncs(repository_location, version):
+    # Checkout sdk-nrf to the desired version.
+    command = "git -C " + repository_location + " checkout " + version
+    process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    if process.returncode:
+        raise subprocess.CalledProcessError(process.returncode, command, output=stdout, stderr=stderr)
+
+    # Call west update command to update all projects and submodules used by sdk-nrf.
+    command = "west update -r"
+    process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    if process.returncode:
+        raise subprocess.CalledProcessError(process.returncode, command, output=stdout, stderr=stderr)
+
+def get_ncs_recommended_version():
+    # Read recommended version saved in the .nrfconnect-recommended-version file.
+    try:
+        with open(chip_root + 'config/nrfconnect/.nrfconnect-recommended-version', 'r') as f:
+            return f.readline().strip()
+    except:
+        raise RuntimeError("Encountered problem when trying to read .nrfconnect-recommended-version file.")
+
+def print_check_version_warning_message(current_version, recommended_version):
+    current_version_message = "WARNING: Your current NCS version (" + current_version + ")"
+    recommended_version_message = "differs from the recommended (" + recommended_version + ")."
+    allowed_message = "Please be aware that it may lead to encountering unexpected problems."
+    update_message = "Consider updating NCS to the recommended version, by calling:"
+    call_command_message = os.path.abspath(__file__) + " --update"
+
+    # Get the longest message lenght, to fit warning frame size.
+    longest_message_len = max([len(current_version_message), len(recommended_version_message), len(allowed_message), len(update_message), len(call_command_message)])
+    
+    # To keep right frame shape the space characters are added to messages shorter than the longest one.
+    print('\33[33m' + (longest_message_len+3)*'#' + '\x1b[0m')
```

Comment:
When stdout is redirected to a file those escape sequences will look very confusing. Please add them only if `sys.stdout.isatty()`